lib/IRGen/IRGenSIL.cpp

```diff
@@ -931,22 +931,24 @@ class IRGenSILFunction :
     }
   }
 
-  // SWIFT_ENABLE_TENSORFLOW
-  // Returns the LLVM function with `funcName`. It must exist.
-  llvm::Function *findFunction(StringRef funcName, SILModule &silModule) {
-    LLVM_DEBUG(llvm::dbgs() << "IRGen for calling " << funcName << "().\n");
-    auto silFn = silModule.findFunction(funcName, SILLinkage::PublicExternal);
-    assert(silFn);
-    llvm::Function *fn = IGM.getAddrOfSILFunction(silFn, NotForDefinition);
-    assert(fn);
-    return fn;
-  }
-
   void checkOk(llvm::Value *status) {
     auto *checkOkFn = IGM.getTFC_CheckOkFn();
     Builder.CreateCall(checkOkFn, {status});
   }
 
+  /// Create an int64 or float array based on `elements` and `isFloat`. On
+  /// return, `*numElts` is set to an int32-typed array size value.  `*array` is
+  /// set to an alloca inst representing the array, where the inst name is set
+  /// to arrayName.
+  ///
+  /// For each value in `elements`, the function runs `convertElt` on it to
+  /// produce a value, to store into the memory buffer allocated at `*array`.
+  template <typename SrcT>
+  void createArrayAndSize(ArrayRef<SrcT> elements, bool isFloat,
+                          const char *arrayName,
+                          std::function<llvm::Value *(SrcT elt)> convertElt,
+                          Address *array, llvm::Value **numElts);
```

Comment:
Is it better to use `*&` here? It's a little safer and has a precedent in this file https://github.com/apple/swift/blob/4a9bd582f1d303c6b733ccbf5a3dc5ec44c42a5e/lib/IRGen/IRGenSIL.cpp#L3549, whereas `**` has no precedent.

Reply:
The google c++ style is to pass an output param by pointer, so that it is more distinguished from input params. For consistency with local style, I will change to passing by reference, but name the var to be `<foo>Out`.

Reply:
Ok! Good to know.